mass_mailing_switzerland/models/res_partner.py

```diff
@@ -105,12 +105,15 @@ def write(self, vals):
                 mailing_contact_vals["opt_out"] = vals["opt_out"]
             if "category_id" in vals:
                 mailing_contact_vals["tag_ids"] = vals["category_id"]
-            if "email" in vals and not vals["email"] and not self.env.context.get(
+            if "email" in vals and vals["email"] and not self.env.context.get(
```

Comment:
I think we want to include the case were we remove the email address
```suggestion
            if "email" in vals and not self.env.context.get(
```